### PR TITLE
[Test] Relax assertion of denial message for service account

### PR DESCRIPTION
**Pull request by ywangd**
Relax the assertion for error message so that v7RestCompat test does not break on non-bwc'able changes.

Relates: #79809

### CODE REVIEW

File: x-pack/plugin/build.gradle [hunk 1]
```diff
@@ -113,6 +113,7 @@ tasks.named("yamlRestTestV7CompatTransform").configure{ task ->
   task.skipTest("indices.freeze/10_basic/Basic", "#70192 -- the freeze index API is removed from 8.0")
   task.skipTest("indices.freeze/10_basic/Test index options", "#70192 -- the freeze index API is removed from 8.0")
   task.skipTest("ml/categorization_agg/Test categorization aggregation with poor settings", "https://github.com/elastic/elasticsearch/pull/79586")
+  task.skipTest("service_accounts/10_basic/Test service account tokens", "mute till we decide whether to make the changes in 7.16")
```
Comment (ywangd) [inline x-pack/plugin/build.gradle:116]: This PR is to ensure the V7 rest compat test can run once the error message is updated consistently across 8.1, 8.0 and 7.16. I will have another PR to remove this `skipTest` once this PR is merged and backported.

**Comment by tvernum** [comment]
Does this need to go into 8.0 or just 7.16?

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/service_accounts/10_basic.yml [hunk 2]
```diff
@@ -85,7 +85,7 @@ teardown:
 
   - match: { "error.type": "security_exception" }
   - match:
-      error.reason: "action [cluster:admin/xpack/security/user/delete] is unauthorized for user [elastic/fleet-server], this action is granted by the cluster privileges [manage_security,all]"
+      error.reason: "/action.\\[cluster:admin/xpack/security/user/delete\\].is.unauthorized.for.*\\[elastic/fleet-server\\].*this.action.is.granted.by.the.cluster.privileges.*/"
```
Comment (tvernum) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/service_accounts/10_basic.yml:88]: I don't understand why the regex needs a `.` in place of a space. Can you explain?

**Review by tvernum — APPROVED** [review]

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/service_accounts/10_basic.yml (hunk 2 above)
Comment (ywangd) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/service_accounts/10_basic.yml:88]: The match regex assertion is compiled with the `COMMENTS` option that ignores whitespaces and permits comments (`#`). 

https://github.com/elastic/elasticsearch/blob/54d69182fdf316375aa3f5df6d1badf970a10851/test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/MatchAssertion.java#L67

We can also assert whitespace with `\ `, e.g. `this\ action\ is\ granted\ by\ the\ cluster\ privileges`. But I think escapes looks ugly and it's not all that important. Many existing yaml tests just use `.`.

**Comment by ywangd** [comment]
> Does this need to go into 8.0 or just 7.16?

For now probably 7.16 is fine since 8.0 is not used for v7RestCompat for 8.1. But I think it's better to update both 8.0 and 7.16 since it is a tiny change and future-proof in case we need restCompat from 8.0 to future versions.

**Pull request merged**